shared/src/current_block.rs

```diff
@@ -156,6 +157,23 @@ where
             .context("failed to get current block")?
             .as_u64())
     }
+
+    async fn block_number_from_tx_hash(&self, hash: H256) -> Option<u64> {
```

Comment:
I think this should propagate the failure to the caller.

```suggestion
    async fn block_number_from_tx_hash(&self, hash: H256) -> Result<u64> {
```

Returning just an `Option` makes the `.context(...)` not do anything either.